src/eventsctrl.py

```diff
@@ -38,14 +51,26 @@ def GET(self, name):
                 params.update(dparams)
 
             criterion = web.input().get('aggregation', '')
-            if not Aggregation.is_allowed(criterion):
-                m = "Aggregation must be one of the following: {}"
-                raise Error(BADPARAMS,
-                            msg=m.format(Aggregation.list_allowed()))
-            aggregation = Aggregation(criterion)
-            aggregation.data = Event.get_for_aggregation(criterion, clause,
-                                                         params)
-            data = aggregation.aggregate()
+
+            query_args = [criterion, clause, params]
+            redis_key = json.dumps(query_args)
+            value = redis_client.get_from_json(redis_key)
+
+            if value:
+                data = json.loads(value)
+            else:
+                if not Aggregation.is_allowed(criterion):
+                    m = (
```

Comment:
(minor) on-letter variables are a bad practice in Python

Reply:
Hmmm, well spotted, actually - I need to re-edit this :thinking: I left the `.format()` in after already converting this to f-strings, so there's no need to actually use a variable... I'll update it now, thanks :+1: